### ISSUE TITLE
Streaming  custom filename

### ISSUE THREAD

**Issue by wjmwjmwb** [channelcat/sanic#925]
Can i customize filename when using response.stream? Now, it depends on url. But my route is immutable.

**Comment by ashleysommer** [comment]
Do you mean, you want to stream a file in response to a http request?
Is the file you are streaming based on the URL of the request?

Does the static files helper work for you in this circumstance?
https://github.com/channelcat/sanic/blob/master/docs/sanic/static_files.md
It has an option to stream the requested files using a streaming response.

**Comment by wjmwjmwb** [comment]
example,my route is "/download/orders". now when i  download file, the file'name is oders.
but i want customize filename,rather than based on the URL.

**Comment by jrocketfingers** [comment]
Forgive me for lacking context, but would something like `Content-Disposition: attachment; filename="filename.jpg"` suffice?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition

**Comment by ashleysommer** [comment]
I just recently came back to this question, and I now understand what you are asking, I apologies, I didn't quite get what you were asking last time I visited.

No, it is not possible to customize the filename of a streaming file, however it should be possible. It is a great feature request, and will be relatively simple to implement.

If you feel up to it, feel free to open a PR with an implementation of this feature.
